Component/LinearSolver/Direct/src/sofa/component/linearsolver/direct/SparseCholeskySolver.inl

```diff
@@ -41,34 +47,69 @@ SparseCholeskySolver<TMatrix,TVector>::~SparseCholeskySolver()
 }
 
 template<class TMatrix, class TVector>
-void SparseCholeskySolver<TMatrix,TVector>::solveT(double * z, double * r)
+void SparseCholeskySolver<TMatrix,TVector>::solveT(double * x, double * b)
 {
     int n = A.n;
-
-    cs_ipvec (n, S->Pinv, r, (double*) &(tmp[0]));	//x = P*b
-
-    cs_lsolve (N->L, (double*) &(tmp[0]));			//x = L\x
-    cs_ltsolve (N->L, (double*) &(tmp[0]));			//x = L'\x/
-
-    cs_pvec (n, S->Pinv, (double*) &(tmp[0]), z);	 //b = P'*x
+    
+    switch( type_perm.getValue().getSelectedId() )
+    {
+        case 0://None->identity
+        default:
+            cs_lsolve (N->L, b );			//x = L\x
+            cs_ltsolve (N->L, b );			//x = L'\x/
+            cs_pvec (n, S->Pinv, b , x );	 //used here to copy, transpose(Pinv) = Id
+            break;
+    
+        case 1://SuiteSparse
+            
+            cs_ipvec (n, S->Pinv,  b, (double*) tmp.data() );	//x = P*b , permutation on rows
+            cs_lsolve (N->L, (double*) tmp.data() );			//x = L\x
+            cs_ltsolve (N->L, (double*) tmp.data() );			//x = L'\x/
+            cs_pvec (n, S->Pinv, (double*) tmp.data() , x );	 //b = P'*x , permutation on columns
+            break;
+
+        case 2://METIS
+
+            cs_ipvec (n, perm.data(),  b , tmp.data() );	//x = P*b , permutation on rowsl;
+            cs_lsolve (N->L, tmp.data() );			//x = L\x
+            cs_ltsolve (N->L, tmp.data() );			//x = L'\x/
+            cs_pvec (n, perm.data() , tmp.data() , x );	 //b = P'*x , permutation on columns
+            break;
+
```

Comment:
Why SuiteSparse and METIS cases cannot be combined? You cannot use a common variable to store the permutation?

Reply:
Pinv is int* and perm is vector<int> , there are some memory issues.
Doable if I manage to resolve them.